app/components/files-list/component.js

```diff
@@ -2,46 +2,37 @@ import Component from '@ember/component';
 import { computed, get } from '@ember/object';
 import { A } from '@ember/array';
 
-export default Component.extend({
+export default class FilesList extends Component {
+  // Named properties passed into the component
+  files;
+  submissionActions;
+
+  editable;
+
+  removeAction;
 
   /**
    * Combine files with any associated actions so they can be displayed
+   * Expect 'files' and 'submissionActions' to be proxy objs
+   *
+   * NOTE: Somehow, making this a Glimmer component breaks this function
    */
-  filesWithActions: computed(
-    'files',
-    'submissionActions',
+  @computed(
+    'files.[]',
     'submissionActions.@each.{key,description}',
-    function () {
-      const actions = get(this, 'submissionActions') || A();
-
-      return get(this, 'files').map((file) => {
-        const id = file.id;
-        const fileActions = actions.filter(a => a.key === id);
-
-        return {
-          file,
-          hasActions: fileActions.length > 0,
-          actions: fileActions
-        };
-      });
-    }
-  ),
+  )
+  get filesWithActions() {
+    const actions = get(this, 'submissionActions') || A();
 
-  actions: {
-    /**
-     * Bubble the remove request up to parent component if `removeAction` is defined
-     * @param {File} file file obj
-     */
-    doRemove(file) {
-      this.removeAction(file);
-    },
+    return get(this, 'files').map((file) => {
```

Comment:
```suggestion
    return this.args.files.map((file) => {
```
I tested this manually and if you make this a glimmer component while just accessing the args here it should work (after getting rid of the computed and the property declarations above).